### ISSUE TITLE
Scrolling behavior

### ISSUE THREAD

**Issue by tim-wgbh** [codevise/pageflow#1004]
Our designer has observed this page transition behavior when trying the hosted version:
https://www.youtube.com/watch?v=OlNmmQcOuTs
(see explanation)

She prefers this version, achieved in shorthand:
https://www.youtube.com/watch?v=LA1WZ1hIb7U

Is there a way to speed up the scrolling? 

Thank you.

**Comment by tf** [comment]
So there is one page per year in that example? The transition duration is hard coded in [CSS](https://github.com/codevise/pageflow/blob/183443f14ab716a98c21cb377a5492ecd925da8d/app/assets/stylesheets/pageflow/page_transitions/crossfade.scss#L4)/[JS](https://github.com/codevise/pageflow/blob/183443f14ab716a98c21cb377a5492ecd925da8d/app/assets/javascripts/pageflow/page_transitions.js#L22) at the moment with no easy way to change it.

Ideally, this would probably rather be a done with some sort of "scrollimation" page type that would still need to be built. Still, if you need a quick solution, you might be able to define a new page transition. There is no official API to do this since page transition CSS depends on internal details of the page DOM structure. But if you needed a quick solution you could probably add something like the following to your app:

```js
// pageflow/application.js

//= require pageflow/base

pageflow.pageTransitions.register('fast_crossfade', {duration: 200});
``` 

Copy [`pageflow/page_transitions/crossfade.scss`](https://github.com/codevise/pageflow/blob/master/app/assets/stylesheets/pageflow/page_transitions/crossfade.scss) to something like:

```scss
// pageflow/application.scss

@import "pageflow/base";

section.fast_crossfade {
  @include transition(0.2s ease);
  // ... leave rest unchanged
}
```

And add translations:

```
# config/locales/en.yml
en:
  pageflow:
    page_transitions:
      fast_crossfade: "Fast Crossfade"
```

Then you should be able to select the new transition on the options tab of a page in the editor.

I haven't tested the code above. Especially the CSS might break with future versions of Pageflow since the used classes are not really part of any public API.

**Comment by tim-wgbh** [comment]
Thank you, this is better, but I think the fundamental problem is that you can only go one page with a scroll movement. Is there a way to enable scrolling multiple pages?

**Comment by tf** [comment]
No, this is not possible at the moment. The main problem is that scrolling is handled by scrollers inside each individual page. When you hit the bottom of such a scroller it triggers the page change animation. Since these scrollers do not provide reliable scroll events while they are being moved, scrolling is disabled while page transitions are in progress. This limits the speed with which one can move through an entry.

As I said before, the "Pageflow-way" of handling this feature request would probably be a separate page type that flicks through a list of images based on the (invisible) position of one such page scroller. 

There's a [guide](https://github.com/codevise/pageflow/blob/master/doc/creating_page_types.md) on how to create page types using React components if you're interested.
